src/lib/Sympa/List.pm

```diff
@@ -842,6 +842,9 @@ sub load {
 
     ## Set of initializations ; only performed when the config is first loaded
     if ($options->{'first_access'}) {
+        if ($robot && (! -d "$Conf::Conf{'home'}/$robot")) {
+            mkdir "$Conf::Conf{'home'}/$robot", 0775
```

Comment:
To keep compatibility with older style path `${expldir}/${listname}`, I think this is better:
```
        if ($robot && (! -d "$Conf::Conf{'home'}/$robot")
            and $robot ne $Conf::Conf{'domain'}) {
            mkdir ...
```
in case some lists have already had `${expldir}/${name}` directories.